src/model/line/mod.rs

```diff
@@ -26,7 +28,11 @@ impl Line {
             "C4 F3 G3 Bb3 C4 Db4 Bb3 Db4 C4 . .",
         ]
         .iter()
-        .map(|line_str| parse_line(line_str).unwrap())
+        .map(|line_str| Self::parse(line_str).unwrap())
         .collect()
```

Comment:
This is not really a valuable change (and is probably harder to read), but it feels a little gross to me to have to manually unwrap inside the map like that. I think you could do this:

```rust
vec![...].iter().map(Self::parse).collect::<Result<Vec<_>>().unwrap()
```

(Collect an iterable of results into a result of a vec, and then unwrap that single result.)

Reply:
Ok no that makes sense, the only thing I had to modify from your example is that for some reason passing `Self::parse` directly as the argument to `.map()` made it expect an argument of type `&&str` to `Self::parse()` (vs just `&str`), not sure I understand why?